### ISSUE TITLE
converter fails to write midi files correctly

### ISSUE THREAD

**Issue by ZM-J** [cuthbertLab/music21#464]
For those who have repeats and changes in `TimeSignature`, converter fails to write them to correct midi files.
Try use case below
```python
    it = corpus.chorales.Iterator(currentNumber=11)
    c = next(it.__iter__())
    c.show() # This score has repeat notation and various TimeSignature, 4/4 -> 3/4 -> 4/4
    ts = c.getTimeSignatures()
    print(len(ts)) # Got 1, expect 2
    c.save('somefile.mid') # Can open somefile.mid from MuseScore 3 directly, but it fails to duplicate the repeating part
    another_c = converter.parse('somefile.mid')
    c.show() # MuseScore 3 fails to show the score
```

**Comment by ZM-J** [comment]
Moreover, traverse `timeSignature` for every measure after `makeMeasures()`:
```python
    it = corpus.chorales.Iterator(currentNumber=11)
    c = next(it.__iter__())
    c.show()
    ms = [m for m in c.makeMeasures() if isinstance(m, stream.Measure)]
    for m in ms:
        nowt = m.getTimeSignatures()
        print(nowt.timeSignature.ratioString)
```
It will output:
```
4/4
4/4
4/4
4/4
4/4
4/4
4/4
4/4
4/4
4/4
4/4
4/4
4/4
4/4
3/4
4/4
4/4
4/4
4/4
4/4
4/4
4/4
4/4
4/4
4/4
4/4
4/4
4/4
4/4
4/4
4/4
4/4
4/4
4/4
4/4
```
3/4 only appears once, which is the problem.

**Comment by mscuthbert** [comment]
Please read the documentation for Stream.getTimeSignatures -- the behavior is correct -- it only gets timeSignatures in the stream itself or failing that, returns 4/4.

https://web.mit.edu/music21/doc/moduleReference/moduleStream.html#music21.stream.Stream.getTimeSignatures

However, for this reason the routine is basically not useful -- it'll be deprecated in the next version of music21.  

For your bug report at first, please give a minimum (less than ten lines and fewer than 10 notes/time signatures/etc.) demonstration that causes the problem.

**Comment by ZM-J** [comment]
> Please read the documentation for Stream.getTimeSignatures -- the behavior is correct -- it only gets timeSignatures in the stream itself or failing that, returns 4/4.
> 
> https://web.mit.edu/music21/doc/moduleReference/moduleStream.html#music21.stream.Stream.getTimeSignatures
> 
> However, for this reason the routine is basically not useful -- it'll be deprecated in the next version of music21.
> 
> For your bug report at first, please give a minimum (less than ten lines and fewer than 10 notes/time signatures/etc.) demonstration that causes the problem.

But in my first test case, where I call `ts = c.getTimeSignatures()`. it only collects one '4/4', but not **ALL** `TimeSignature` objects mentioned in your documentation.
The chorale generated by iterator this project provides can tell the problem clearly.

**Comment by jacobtylerwalls** [comment]
Take another look at your chorale -- it has one 3/4 time signature and no repeats. 

The expected result of getTimeSignatures() on the entire chorale is one instance of 3/4.
The result of getTimeSignature() on each individual measure would be, in most cases 4/4, except the pickup measure that contains the 3/4.

I believe you showed us the expected result in both cases. :-)

Also the MIDI comments are hard to follow because you have a "c" variable that is a stream and then a "c" variable that is a converter most likely, but we never saw what you parsed into it.

**Comment by jacobtylerwalls** [comment]
I think you need to flatten the score to get the result you want:

```
it = corpus.chorales.Iterator(currentNumber=11)
c = next(it.__iter__())
ts = c.flat.getTimeSignatures()
len(ts)
>>> 27
```

**Comment by jacobtylerwalls** [comment]
9 parts * 3 time signatures each. Ultimately it sounds like you wanted this:

```
set([sig.ratioString for sig in ts])
{'4/4', '3/4'}
```

**Comment by jacobtylerwalls** [comment]
heya @ZM-J I think I understand your original issue a little better now. Are you requesting that we account for repeats when writing out MIDI files? It looks to me like m21 doesn't support that today, and you have a fairly small test case for that, which is good. I suggest renaming this ticket to just say "Account for repeats when exporting MIDI"

**Comment by ZM-J** [comment]
Exactly.
I don't know what this project may regard repeats as. I just purely think that since the repeats plays twice in MuseScore (and in midi files exported by MuseScore), it should be the same as those exported by Music21.

**Comment by ZM-J** [comment]
> heya @ZM-J I think I understand your original issue a little better now. Are you requesting that we account for repeats when writing out MIDI files? It looks to me like m21 doesn't support that today, and you have a fairly small test case for that, which is good. I suggest renaming this ticket to just say "Account for repeats when exporting MIDI"

Exactly.
I don't know what this project may regard repeats as. I just purely think that since the repeats plays twice in MuseScore (and in midi files exported by MuseScore), it should be the same as those exported by Music21.
